youtube_dl/extractor/tvp.py

```diff
@@ -1,139 +1,247 @@
-# -*- coding: utf-8 -*-
-from __future__ import unicode_literals
+# -*- encoding: utf-8 -*-
+from .common import InfoExtractor, ExtractorError
 
-import re
+VIDEO_LISTING_URL = ('http://www.api.v3.tvp.pl/shared/listing.php'
+                     '?dump=json&direct=true&count=-1&parent_id={id}')
+META_URL = 'http://www.tvp.pl/shared/video_data.php?dump=json&video_id={id}'
+TOKENIZER_URL = 'http://www.tvp.pl/shared/cdn/tokenizer_v2.php?object_id={id}'
+FILE_INFO_URL = 'http://www.tvp.pl/pub/stat/videofileinfo?video_id={id}'
+IGNORED_MIMETYPES = 'application/vnd.ms-ss', 'application/x-mpegurl'
 
-from .common import InfoExtractor
+
+class TvpApi:
+
+    def __init__(self, ie):
+        """:type ie: InfoExtractor"""
+        self.ie = ie
+
+    def listing(self, id):
+        json = self._get_json(VIDEO_LISTING_URL, id)
+        return json
+
+    def meta(self, id):
+        json = self._get_json(META_URL, id)
+        return json
+
+    def info(self, id):
+        json = self._get_json(FILE_INFO_URL, id)
+        return json
```

Comment:
Code duplication in these three methods. Again they are only used once, so there is really no need in them.